### ISSUE TITLE
[loki-stack] Unable to install

### ISSUE THREAD

**Issue by Dunge** [prometheus-community/helm-charts#3069]
### Describe the bug a clear and concise description of what the bug is.

I have kube-prometheus-stack installed with metrics and grafana dashboards functional. I want to install loki to add functionality to see my pods logs. I think loki-stack is the best compatible helm chart that also install promtail. Looking at the value files, there doesn't seem like there's anything needed to configure.

Unfortunately, it crash on launch.

### What's your helm version?

version.BuildInfo{Version:"v3.8.2", GitCommit:"6e3701edea09e5d55a8ca2aae03a68917630e91b", GitTreeState:"clean", GoVersion:"go1.17.5"}

### What's your kubectl version?

Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6", GitCommit:"ad3338546da947756e8a88aa6822e9c11e7eac22", GitTreeState:"clean", BuildDate:"2022-04-14T08:49:13Z", GoVersion:"go1.17.9", Compiler:"gc", Platform:"windows/amd64"} Server Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.1", GitCommit:"8f94681cd294aa8cfd3407b8191f6c70214973a4", GitTreeState:"clean", BuildDate:"2023-01-27T20:26:13Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}

### Which chart?

loki-stack

### What's the chart version?

2.9.9

### What happened?

promtail pods runs succesfully, loki-0 pod CrashLoopBackOff with following error log:

```
level=info ts=2023-02-24T07:52:23.973846262Z caller=main.go:103 msg="Starting Loki" version="(version=2.6.1, branch=HEAD, revision=6bd05c9a4)"
level=info ts=2023-02-24T07:52:23.974285735Z caller=server.go:288 http=[::]:3100 grpc=[::]:9095 msg="server listening on addresses"
level=info ts=2023-02-24T07:52:23.975431507Z caller=modules.go:736 msg="RulerStorage is not configured in single binary mode and will not be started."
level=warn ts=2023-02-24T07:52:23.978586546Z caller=experimental.go:20 msg="experimental feature in use" feature="In-memory (FIFO) cache - chunksfifocache"
level=info ts=2023-02-24T07:52:23.978956446Z caller=table_manager.go:252 msg="query readiness setup completed" duration=1.473µs distinct_users_len=0
level=info ts=2023-02-24T07:52:23.978983685Z caller=shipper.go:124 msg="starting index shipper in RW mode"
level=info ts=2023-02-24T07:52:23.979051282Z caller=table_manager.go:134 msg="uploading tables"
level=info ts=2023-02-24T07:52:23.98001867Z caller=shipper_index_client.go:79 msg="starting boltdb shipper in RW mode"
level=info ts=2023-02-24T07:52:23.98111501Z caller=modules.go:761 msg="RulerStorage is nil.  Not starting the ruler."
level=info ts=2023-02-24T07:52:23.981718066Z caller=table_manager.go:167 msg="handing over indexes to shipper"
level=info ts=2023-02-24T07:52:23.985863701Z caller=worker.go:112 msg="Starting querier worker using query-scheduler and scheduler ring for addresses"
level=info ts=2023-02-24T07:52:23.989253896Z caller=module_service.go:82 msg=initialising module=server
level=info ts=2023-02-24T07:52:23.989432381Z caller=module_service.go:82 msg=initialising module=query-frontend-tripperware
level=info ts=2023-02-24T07:52:23.989453249Z caller=module_service.go:82 msg=initialising module=memberlist-kv
level=error ts=2023-02-24T07:52:23.989585575Z caller=loki.go:386 msg="module failed" module=memberlist-kv error="invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=error ts=2023-02-24T07:52:23.989727317Z caller=loki.go:386 msg="module failed" module=querier error="failed to start querier, because it depends on module ring, which has failed: invalid service state: Failed, expected: Running, failure: failed to start ring, because it depends on module memberlist-kv, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=error ts=2023-02-24T07:52:23.989900367Z caller=loki.go:386 msg="module failed" module=ingester error="failed to start ingester, because it depends on module store, which has failed: invalid service state: Failed, expected: Running, failure: failed to start store, because it depends on module memberlist-kv, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=error ts=2023-02-24T07:52:23.98999053Z caller=loki.go:386 msg="module failed" module=usage-report error="failed to start usage-report, because it depends on module memberlist-kv, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=error ts=2023-02-24T07:52:23.990085587Z caller=loki.go:386 msg="module failed" module=ring error="failed to start ring, because it depends on module memberlist-kv, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=error ts=2023-02-24T07:52:23.990181431Z caller=loki.go:386 msg="module failed" module=store error="failed to start store, because it depends on module memberlist-kv, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=error ts=2023-02-24T07:52:23.990344372Z caller=loki.go:386 msg="module failed" module=ingester-querier error="failed to start ingester-querier, because it depends on module memberlist-kv, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=error ts=2023-02-24T07:52:23.990503905Z caller=loki.go:386 msg="module failed" module=compactor error="failed to start compactor, because it depends on module memberlist-kv, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=error ts=2023-02-24T07:52:23.990603705Z caller=loki.go:386 msg="module failed" module=distributor error="failed to start distributor, because it depends on module memberlist-kv, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=error ts=2023-02-24T07:52:23.990709554Z caller=loki.go:386 msg="module failed" module=query-frontend error="failed to start query-frontend, because it depends on module usage-report, which has failed: invalid service state: Failed, expected: Running, failure: failed to start usage-report, because it depends on module memberlist-kv, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=error ts=2023-02-24T07:52:23.99082232Z caller=loki.go:386 msg="module failed" module=query-scheduler error="failed to start query-scheduler, because it depends on module memberlist-kv, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: service &{0xc0008cf9a0 { true 10000000000 4 30000000000 200000000 3 30000000000 0 true  7946  false [loki-memberlist] 1000000000 60000000000 10 false 0 300000000000 5000000000 0 {[] 7946 5000000000 5000000000 false 0xc00009aa50 loki false {    false}} 0xc00009aa50 loki [{ringDesc 0xe43cc0} {}]} 0xc00091a1e0 0xc00009aa50 0xc0001cf650 {{} [0 1 0]} <nil> <nil> {0 0} map[] map[ringDesc:{ringDesc 0xe43cc0} usagestats.jsonCodec:{}] {0 0} map[] map[] {0 0} [] 0 [] 0 0 {0 0} map[] 0xc00013ca20 0xc000164840 0xc000164900 0xc0001649c0 0xc000164a80 0xc000164cc0 0xc000164b40 0xc000164d80 0xc000164c00 0xc000488040 0xc000488080 0xc000164f00 0xc000164fc0 0xc000165140 0xc000165080 0xc000240588 0xc0001e0bd0 0xc000240578 0xc000240580 0xc0004880c0 0xc000488100 10} failed: failed to create memberlist: Failed to get final advertise address: no private IP address found, and explicit IP not provided"
level=info ts=2023-02-24T07:52:23.989824Z caller=module_service.go:114 msg="module stopped" module=query-frontend-tripperware
level=info ts=2023-02-24T07:52:23.992347249Z caller=modules.go:1090 msg="server stopped"
level=info ts=2023-02-24T07:52:23.992375764Z caller=module_service.go:114 msg="module stopped" module=server
level=info ts=2023-02-24T07:52:23.992390572Z caller=loki.go:375 msg="Loki stopped"
failed services
github.com/grafana/loki/pkg/loki.(*Loki).Run
 /src/loki/pkg/loki/loki.go:421
main.main
 /src/loki/cmd/loki/main.go:105
runtime.main
 /usr/local/go/src/runtime/proc.go:255
runtime.goexit
 /usr/local/go/src/runtime/asm_amd64.s:1581
level=error ts=2023-02-24T07:52:23.992474923Z caller=log.go:103 msg="error running loki" err="failed services\ngithub.com/grafana/loki/pkg/loki.(*Loki).Run\n\t/src/loki/pkg/loki/loki.go:421\nmain.main\n\t/src/loki/cmd/loki/main.go:105\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:255\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581"
```

### What you expected to happen?

pod to launch

### Enter the command that you execute and failing/misfunctioning.

helm install loki grafana/loki-stack --version 2.9.9

**Comment by Dunge** [comment]
I tried with the `loki-simple-scalable` chart too based on [this guide](https://grafana.com/docs/loki/latest/installation/helm/install-monolithic/) (since I don't have cloud object storage available) and I have the same errors on all loki-read and loki-write pods.

The kubernetes cluster is hosted by microk8s and which I believe use calico for the network stack.

I saw this issue https://github.com/grafana/loki/issues/6370 that talks about the problem.

Unfortunately the helm values can't be set the same way with every helm charts.
